### ISSUE TITLE
Possibility to edit style on user coded figure

### ISSUE THREAD

**Issue by yalap13** [GraphingLib/gl-style-editor#46]
Add a way to input a python file with the code for a figure and make it possible to edit the style on the figure, allowing the user to see the style in its use-case.

**Comment by Oddball777** [comment]
Proof of concept. As long as there's a graphinglib figure created in the file, it will work. Can handle multiple figures, let's the user choose. Overrides the .show() and .save() methods so that they don't pop up when running the script.

```python

import sys

import graphinglib as gl
import matplotlib.pyplot as plt
from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
from PyQt5.QtWidgets import (
    QApplication,
    QFileDialog,
    QInputDialog,
    QMessageBox,
    QPushButton,
    QVBoxLayout,
    QWidget,
)


def dummy_show(*args, **kwargs):
    pass


def dummy_save(*args, **kwargs):
    pass


class MplCanvas(FigureCanvas):
    def __init__(self, fig):
        self.fig = fig
        super(MplCanvas, self).__init__(self.fig)


class MainWindow(QWidget):
    def __init__(self):
        super().__init__()
        self.layout = QVBoxLayout()
        self.button = QPushButton("Load Python File")
        self.button.clicked.connect(self.loadPythonFile)
        self.layout.addWidget(self.button)
        self.setLayout(self.layout)
        self.setGeometry(200, 200, 700, 500)
        self.setWindowTitle("Matplotlib Figure Viewer")

    def loadPythonFile(self):
        filepath, _ = QFileDialog.getOpenFileName(
            self, "Open Python file", "", "Python Files (*.py)"
        )
        if filepath:
            self.executePythonFile(filepath)

    def executePythonFile(self, filepath):
        # Clear existing widgets except the button
        for i in reversed(range(self.layout.count())):
            widgetToRemove = self.layout.itemAt(i).widget()
            if widgetToRemove is not self.button:
                self.layout.removeWidget(widgetToRemove)
                widgetToRemove.setParent(None)

        original_show = gl.Figure.show
        original_save = gl.Figure.save
        original_multi_show = gl.MultiFigure.show
        original_multi_save = gl.MultiFigure.save
        gl.Figure.show = dummy_show
        gl.Figure.save = dummy_save
        gl.MultiFigure.show = dummy_show
        gl.MultiFigure.save = dummy_save

        # Execute the script
        namespace = {"gl": gl, "__builtins__": __builtins__}
        with open(filepath) as file:
            code = compile(file.read(), filepath, "exec")
            exec(code, namespace, namespace)

        gl.Figure.show = original_show
        gl.Figure.save = original_save
        gl.MultiFigure.show = original_multi_show
        gl.MultiFigure.save = original_multi_save
        # Check for figures in the namespace and display them
        figures = {}
        for name, var in namespace.items():
            if isinstance(var, gl.Figure) or isinstance(var, gl.MultiFigure):
                figures[name] = var

        # Popup to ask user which figure to display
        if len(figures) > 1:
            chosen = self.chooseFigure(figures.keys())
        else:
            chosen = list(figures.keys())[0]

        if chosen is not None:
            fig = figures[chosen]
            if isinstance(fig, gl.MultiFigure):
                fig._prepare_multi_figure()
            else:
                fig._prepare_figure()
            self.displayFigure(fig._figure)

    def displayFigure(self, fig):
        canvas = MplCanvas(fig)
        self.layout.addWidget(canvas)
        canvas.draw()

    def chooseFigure(self, figures):
        # Popup to ask user which figure to display (figures is a list of figure names)
        figure_names = list(figures)
        chosen, ok = QInputDialog.getItem(
            self, "Choose Figure", "Select a figure:", figure_names, editable=False
        )
        if ok and chosen:
            return chosen
        else:
            return None


if __name__ == "__main__":
    app = QApplication(sys.argv)
    main = MainWindow()
    main.show()
    sys.exit(app.exec_())

```

**Comment by Oddball777** [comment]
Closed with PR #52.
